### PR TITLE
Accept SENSU_API_KEY in addition to SENSU_USER and SENSU_PASSWORD

### PR DESCRIPTION
**Pull request by mhuysamen**

### CODE REVIEW

**Comment by kmf** [comment]
@calebhailey / @jspaleta anyone?

**Comment by jspaleta** [comment]
Testing now,

Question for you... are you using this script locally or as a github action?

If your using it as a github action we'll need to expose the new env_var as a settable option for the github action.

File: sensuflow.sh [hunk 1]
```diff
@@ -98,12 +103,15 @@ if [ $status -lt 200 ] || [ $status -ge 400 ]; then
...
-if [[ $VERBOSE ]]; then echo "Checking Sensu auth credentials"; fi
-status=$(${curl_command} --connect-timeout 30 -k -o /dev/null -s -w "%{http_code}" -X GET "${SENSU_BACKEND_URL}/auth/test" -u "${SENSU_USER}:${SENSU_PASSWORD}")
-if [ $status -lt 200 ] || [ $status -ge 400 ]; then
-	echo "Sensu auth failed"
-	echo "Probe of "$SENSU_BACKEND_URL/auth/test" returned status code: $status"
-	exit 1
+
+if [ -z "$SENSU_API_KEY" ]; then # Skip auth test if API KEY is specified
+  if [[ $VERBOSE ]]; then echo "Checking Sensu auth credentials"; fi
+  status=$(${curl_command} --connect-timeout 30 -k -o /dev/null -s -w "%{http_code}" -X GET "${SENSU_BACKEND_URL}/auth/test" -u "${SENSU_USER}:${SENSU_PASSWORD}")
```
Comment (jspaleta) [inline sensuflow.sh:109]: This test doesn't have to be skipped, it just needs to be modified to pass the api key as an http header.

**Review by jspaleta — CHANGES_REQUESTED** [review]

File: sensuflow.sh (hunk 1 above)
Comment (jspaleta) [reply sensuflow.sh:109]: Nope I'm wrong, the auth/test endpoint is only valid for user/pass.
Okay this looks good as it is. I'm going to do a follow up pr to refactor the github action definition that uses this script to allow for api key use

**Review by jspaleta — APPROVED** [review]

**Pull request merged**